org.scala-ide.sdt.core.tests/src/org/scalaide/core/semantichighlighting/classifier/RegionParser.scala

```diff
@@ -2,10 +2,68 @@ package org.scalaide.core.semantichighlighting.classifier
 
 import org.eclipse.jface.text.IRegion
 import org.eclipse.jface.text.Region
+import scala.annotation.tailrec
+
+import language.implicitConversions
 
 object RegionParser {
 
   /**
+   * This class represents a substring with an optional prefix and suffix.
+   */
+  final case class EmbeddedSubstr(str: String, prefix: String = "", suffix: String = "") {
+    private[RegionParser] val searchString = prefix + str + suffix
+  }
+
+  object EmbeddedSubstr {
+    implicit def wrapAsEmbeddedSubstring(str: String) = EmbeddedSubstr(str)
+  }
+
+
+  /**
+   * Extracts the regions marked by the given substrings.
+   *
+   * @example
+   * In the easiest case this function just extracts substrings (note that there is an implicit conversion
+   * from [[[String]]] to [[[org.scalaide.core.semantichighlighting.classifier.RegionParser.EmbeddedSubstr]]]):
+   * {{{
+   * scala> RegionParser.substrRegions("x + f(x)", "x")
+   * res0: Map(offset: 0, length: 1 -> EmbeddedSubstr(x,,), offset: 6, length: 1 -> EmbeddedSubstr(x,,))
+   * }}}
+   *
+   * @example
+   * Extract occurrences of `x` when surrounded by parenthesis:
+   * {{{
+   *  scala> substrRegions("x + f(x)", EmbeddedSubstr("x", "(", ")"))
+   *  res0: Map(offset: 6, length: 1 -> EmbeddedSubstr(x,(,)))
+   * }}}
+   *
+   * @example
+   * Here we extract both occurrences of `x` separately:
+   * {{{
+   * scala> RegionParser.substrRegions("x + f(x)", EmbeddedSubstr("x", suffix = " "), EmbeddedSubstr("x", "(", ")"))
+   * res0: Map(offset: 0, length: 1 -> EmbeddedSubstr(x,, ), offset: 6, length: 1 -> EmbeddedSubstr(x,(,)))
+   * }}}
+   */
+  def substrRegions(test: String, substrs: EmbeddedSubstr*): Map[IRegion, EmbeddedSubstr] = {
```

Comment:
In my opinion there are several advantages of `substrRegions` compared to `delimitedRegions`:

1. There is not need to obscure the test cases with markers.
2. You don't need templates (see AbstractSymbolClassifierTest.checkSymbolClassification) that result in duplication.
3. It's simpler.

I didn't think this through, because I don't think it's worth the effort anyway, but my suspicion is that `AbstractSymbolClassifierTest.checkSymbolClassification` and all tests that use it could get rid of the `locationTemplate` parameter and therefore a significant amount of duplication if they where refactored to use `substrRegions`.